### DISCUSSION TITLE
validate string input if it's a valid enum

### DISCUSSION THREAD

**Discussion by khurram-ansari** [0no-co/gql.tada#308] [Q&A]
I have a countriesEnum in my schema and I am using graphql.env.ts file.

`export type TCountries = ReturnType<typeof graphql.scalar<'CountriesEnum'>>;`

I want to validate a string input against it.

I came across the example in docs regarding enums validation using the `scalar` function but I am unable to understand, is this what I should use to achieve the validation?

**Comment by kitten** [comment]
Depends on what you mean by validation.

Validation itself in GraphQL is not done on the client-side but the server-side and `gql.tada` is a library for authoring documents that are typed, for type validation.

So, if you have an arbitrary string, then you can either only cast it to the GraphQL enum type, send it as part of your mutation, and let your API validate it, or write code that validates it on the client-side
